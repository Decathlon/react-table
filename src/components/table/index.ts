export { default as Table } from "./table";
export * from "./table";

export { default as Row } from "./row";
export * from "./row";

export { default as RowSpan } from "./row-span";
export * from "./row-span";

export { default as Cell } from "./cell";
export * from "./cell";

export { default as ElementaryTable } from "./elementary-table";
export * from "./elementary-table";
