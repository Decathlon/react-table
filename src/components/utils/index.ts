export * from "./table";
export * from "./table-selection";
export * from "./common";

export { default as shallowEqual } from "./shallowEqual";
