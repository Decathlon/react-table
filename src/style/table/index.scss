@import "cell";
@import "row";
@import "row-span";
