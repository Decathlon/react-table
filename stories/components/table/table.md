# Table

Default table for your application.

Able to handle customized content, different row/column size, and child rows.
