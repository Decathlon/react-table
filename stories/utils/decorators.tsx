import * as React from "react";
import { MuiThemeProvider, createMuiTheme } from "@material-ui/core/styles";

export const blueDkt = "#0082c3";

const muiDktTheme = createMuiTheme({
  typography: {
    fontFamily: "Roboto Condensed"
  },
  palette: {
    primary: {
      light: blueDkt,
      dark: blueDkt,
      main: blueDkt
    },
    secondary: {
      light: blueDkt,
      dark: blueDkt,
      main: blueDkt
    }
  }
});

/**
 * The MUI theme provider wrapper
 *
 * @param {() => JSX.Element} attributes a function to render the child component.
 *
 * @return {JSX.Element} The child wrapped by the MUI theme provider.
 */
export function withThemeProvider(renderChild: () => JSX.Element): JSX.Element {
  return <MuiThemeProvider theme={muiDktTheme}>{renderChild()}</MuiThemeProvider>;
}
