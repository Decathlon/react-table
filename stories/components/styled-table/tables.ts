import { IRow } from "../../../src/components/table/row";
import HeaderCell from "../../../src/components/styled-table/header-cell";

const indicatorNames = [
  "CA_TTC_OMNICANAL",
  "PROG_CA_TTC",
  "MARGIN_RATE",
  "HOURS",
  "FEES_PROGRESSION",
  "TURNOVER",
  "RATE_SHRINKAGE",
  "PERSONAL_FEES",
  "SQUARE_METERS",
  "SURFACE_COST",
  "TOTAL_FEES",
  "CONTRIBUTION",
  "OVERHEADS",
  "ADVISED_TURNOVER",
  "CA_TTC_OMNICANALF",
  "PROG_CA_TTCF",
  "MARGIN_RATEF",
  "HOURSF",
  "FEES_PROGRESSIONF",
  "TURNOVERF",
  "RATE_SHRINKAGEF",
  "PERSONAL_FEESF",
  "SQUARE_METERSF",
  "SURFACE_COSTF",
  "TOTAL_FEESF",
  "CONTRIBUTIONF",
  "OVERHEADSF",
  "ADVISED_TURNOVERF",
];

const text = "Lorem Ipsum is that it has a more-or-less normal";

export function getTable(cells = {}) {
  const weeks = Array.from(Array(52), (_, rowIndex) => ({
    id: rowIndex < 9 ? `0${rowIndex + 1}` : `${rowIndex + 1}`,
    value: `W${rowIndex + 1}${rowIndex === 3 ? text : ""}`,
  }));
  const headerRow: IRow = {
    id: "headers",
    isHeader: true,
    size: 126,
    cells: [
      {
        id: "indicators",
        value: `Indicateurs ${text}`,
      },
    ],
  };
  weeks.forEach((week) =>
    headerRow.cells.push({
      id: week.id,
      value: week.value,
      cellContent: HeaderCell,
      cellContentProps: {
        title: "2019",
        isCurrent: week.value === "W12",
        subtitle: "1 éphéméride",
        value: week.value,
      },
    })
  );
  const rows: IRow[] = [headerRow];

  indicatorNames.forEach((indicatorName) => {
    const row: IRow = {
      id: indicatorName,
      cells: [
        {
          id: indicatorName,
          value: indicatorName,
        },
      ],
    };
    weeks.forEach((week) => {
      row.cells.push({
        id: `${indicatorName}-${week.value}`,
        value: "0.00",
      });
    });
    rows.push(row);
  });

  Object.keys(cells).forEach((row) => {
    Object.keys(cells[row]).forEach((cell) => {
      rows[row].cells[cell] = cells[row][cell];
    });
  });

  return { id: "styled-tabel", rows };
}
