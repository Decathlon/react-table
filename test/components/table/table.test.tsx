/// <reference path="../../typings/tests-entry.d.ts" />
import { createRenderer } from "react-test-renderer/shallow";

import { mount } from "enzyme";
import Table, { IState } from "../../../src/components/table/table";
import { simpleTable, tableWithSubItems, subRows, subMiam, generateTable } from "../../../stories/utils/tables";
import { withThemeProvider } from "../../../stories/utils/decorators";
import { getIndexesIdsMapping } from "../../../src/components/utils/table";

const defaultProps = {
  id: "table-foo",
  rows: simpleTable({})
};

const table2Levels = tableWithSubItems({ firstSubRows: subRows({}) });

const table3Levels = tableWithSubItems({
  firstSubRows: subRows({ subsubRows: subMiam })
});

const tableWithSpan = tableWithSubItems({
  firstSubRows: subRows({}),
  secondSubRows: subRows({})
});

describe("Table component", () => {
  test("should render the default table", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Table {...defaultProps} isSelectable={false} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a Table with 2 levels", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Table {...defaultProps} rows={table2Levels} isSelectable={false} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a Table with 3 levels", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Table {...defaultProps} rows={table3Levels} isSelectable={false} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a Table with spans", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Table {...defaultProps} isSpan rows={tableWithSpan} isSelectable={false} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a selectable Table", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Table {...defaultProps} isSpan rows={tableWithSpan} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a responsive Table", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<Table {...defaultProps} isVirtualized rows={tableWithSpan} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should render a virtualized Table", () => {
    const shallowRenderer = createRenderer();
    shallowRenderer.render(
      <Table
        {...defaultProps}
        isVirtualized
        virtualizerProps={{
          height: 300,
          width: 400,
          initialScroll: {
            columnIndex: 3,
            rowIndex: 4
          }
        }}
        rows={tableWithSpan}
      />
    );
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  test("should open sub rows", () => {
    const props = {
      id: "foo",
      rows: subRows({ subsubRows: subMiam })
    };
    const wrapper = mount(withThemeProvider(() => <Table {...props} />));
    const instance: Table = wrapper.find(Table).instance() as Table;

    let expectedState: IState = {
      rowsLength: 2,
      indexesMapping: {
        absolute: {
          "0": { index: 0, parentIndex: null },
          "1": { index: 1, parentIndex: null }
        },
        relative: { "0": { index: 0 }, "1": { index: 1 } }
      },
      columnsIndexesIdsMapping: getIndexesIdsMapping(props.rows[0].cells),
      openedTrees: {},
      fixedRowsIndexes: []
    };

    expect(instance.state).toEqual(expectedState);
    // @ts-ignore private method
    instance.onRowOpen({ rowIndex: 0, columnIndex: 2 });
    expectedState = {
      indexesMapping: {
        absolute: {
          "0": { index: 0, parentIndex: null },
          "1": { index: 0, parentIndex: 0 },
          "2": { index: 1, parentIndex: null }
        },
        relative: {
          "0": { index: 0, subItems: { "0": { index: 1 } } },
          "1": { index: 2 }
        }
      },
      columnsIndexesIdsMapping: getIndexesIdsMapping(props.rows[0].cells),
      openedTrees: { 0: { columnIndex: 2, rowIndex: 0 } },
      rowsLength: 3,
      fixedRowsIndexes: []
    };
    expect(instance.state).toEqual(expectedState);
    // @ts-ignore private method
    instance.onRowOpen({ rowIndex: 1, columnIndex: 2 });
    expectedState = {
      indexesMapping: {
        absolute: {
          "0": { index: 0, parentIndex: null },
          "1": { index: 0, parentIndex: 0 },
          "2": { index: 1, parentIndex: null },
          "3": { index: 0, parentIndex: 2 }
        },
        relative: {
          "0": { index: 0, subItems: { "0": { index: 1 } } },
          "1": { index: 2, subItems: { "0": { index: 3 } } }
        }
      },
      columnsIndexesIdsMapping: getIndexesIdsMapping(props.rows[0].cells),
      openedTrees: {
        0: { columnIndex: 2, rowIndex: 0 },
        1: { columnIndex: 2, rowIndex: 1 }
      },
      rowsLength: 4,
      fixedRowsIndexes: []
    };
    expect(instance.state).toEqual(expectedState);
  });

  test("should close sub row", () => {
    const props = {
      id: "foo",
      rows: subRows({ subsubRows: subMiam })
    };
    const wrapper = mount(withThemeProvider(() => <Table {...props} />));
    const instance: Table = wrapper.find(Table).instance() as Table;
    // tested
    // @ts-ignore private method
    instance.onRowOpen({ rowIndex: 0, columnIndex: 2 });
    // @ts-ignore private method
    instance.onRowClose({ rowIndex: 0, columnIndex: 2 });
    const expectedState: IState = {
      rowsLength: 2,
      indexesMapping: {
        absolute: {
          "0": { index: 0, parentIndex: null },
          "1": { index: 1, parentIndex: null }
        },
        relative: { "0": { index: 0 }, "1": { index: 1 } }
      },
      columnsIndexesIdsMapping: getIndexesIdsMapping(props.rows[0].cells),
      openedTrees: {},
      fixedRowsIndexes: []
    };
    expect(instance.state).toEqual(expectedState);
  });

  test("should call the scrollToColumnIndex virtualizer method (scroll to column index)", () => {
    const props = generateTable(50, 50);
    const wrapper = mount(
      withThemeProvider(() => <Table {...props} isVirtualized virtualizerProps={{ height: 300, width: 400 }} />)
    );
    const instance: Table = wrapper.find(Table).instance() as Table;

    // @ts-ignore
    instance.virtualizer.current.scrollToColumnIndex = jest.fn();
    instance.goToColumnIndex(25);
    // @ts-ignore
    expect(instance.virtualizer.current.scrollToColumnIndex).toBeCalledWith(25);

    instance.goToColumnIndex(-30);
    // @ts-ignore
    expect(instance.virtualizer.current.scrollToColumnIndex).toBeCalledWith(0);

    instance.goToColumnIndex(100);
    // @ts-ignore
    expect(instance.virtualizer.current.scrollToColumnIndex).toBeCalledWith(49);
  });

  test("should call the scrollToRowIndex virtualizer method(scroll to row index)", () => {
    const props = generateTable(50, 50);
    const wrapper = mount(
      withThemeProvider(() => <Table {...props} isVirtualized virtualizerProps={{ height: 300, width: 400 }} />)
    );
    const instance: Table = wrapper.find(Table).instance() as Table;
    // @ts-ignore
    instance.virtualizer.current.scrollToRowIndex = jest.fn();
    instance.goToRowIndex(25);
    // @ts-ignore
    expect(instance.virtualizer.current.scrollToRowIndex).toBeCalledWith(25);

    instance.goToRowIndex(-10);
    // @ts-ignore
    expect(instance.virtualizer.current.scrollToRowIndex).toBeCalledWith(0);

    instance.goToRowIndex(100);
    // @ts-ignore
    expect(instance.virtualizer.current.scrollToRowIndex).toBeCalledWith(49);
  });

  test("should call the scrollToColumnIndex virtualizer method (scroll to column id)", () => {
    const props = generateTable(50, 50);
    const wrapper = mount(
      withThemeProvider(() => <Table {...props} isVirtualized virtualizerProps={{ height: 300, width: 400 }} />)
    );
    const instance: Table = wrapper.find(Table).instance() as Table;

    // @ts-ignore
    instance.virtualizer.current.scrollToColumnIndex = jest.fn();
    instance.goToColumnId("(0,25)-0");
    // @ts-ignore
    expect(instance.virtualizer.current.scrollToColumnIndex).toBeCalledWith(25);
    // @ts-ignore
    instance.virtualizer.current.scrollToColumnIndex.mockReset();
    instance.goToColumnId("(0,300)-0");
    // @ts-ignore
    expect(instance.virtualizer.current.scrollToColumnIndex).not.toBeCalled();
  });

  test("should call the getColumnIndex method", () => {
    const props = generateTable(50, 50);
    const wrapper = mount(
      withThemeProvider(() => <Table {...props} isVirtualized virtualizerProps={{ height: 300, width: 400 }} />)
    );
    const instance: Table = wrapper.find(Table).instance() as Table;

    expect(instance.getColumnIndex("(0,25)-0")).toEqual(25);
    expect(instance.getColumnIndex("notexist")).toBeFalsy();
  });
});
