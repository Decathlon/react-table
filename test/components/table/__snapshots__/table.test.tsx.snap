// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should render a Table with 2 levels 1`] = `
<ElementaryTable
  elevatedColumnIndexes={Object {}}
  elevatedRowIndexes={Object {}}
  id="table-foo"
  indexesMapping={
    Object {
      "absolute": Object {
        "0": Object {
          "index": 0,
          "parentIndex": null,
        },
        "1": Object {
          "index": 1,
          "parentIndex": null,
        },
        "2": Object {
          "index": 2,
          "parentIndex": null,
        },
      },
      "relative": Object {
        "0": Object {
          "index": 0,
        },
        "1": Object {
          "index": 1,
        },
        "2": Object {
          "index": 2,
        },
      },
    }
  }
  onRowClose={[Function]}
  onRowOpen={[Function]}
  openedTrees={Object {}}
  rows={
    Array [
      Object {
        "cells": Array [
          Object {
            "id": "wawoo",
            "value": "Wawooo!",
          },
          Object {
            "id": "wah",
            "value": "Wah!",
          },
          Object {
            "id": "cool",
            "value": "Cool!",
          },
        ],
        "id": "front",
        "isHeader": true,
        "isSelectable": false,
      },
      Object {
        "cells": Array [
          Object {
            "id": "react",
            "subItems": Array [
              Object {
                "cells": Array [
                  Object {
                    "id": "storybook",
                    "value": "Storybook",
                  },
                  Object {
                    "id": "jest",
                    "value": "Jest",
                  },
                  Object {
                    "id": "Enzym",
                    "subItems": Array [],
                    "value": "enzyme",
                  },
                ],
                "id": "foo",
              },
              Object {
                "cells": Array [
                  Object {
                    "id": "docker",
                    "value": "Docker",
                  },
                  Object {
                    "id": "doccomp",
                    "value": "Docker compose",
                  },
                  Object {
                    "id": "Sentry",
                    "subItems": Array [],
                    "value": "Sentry",
                  },
                ],
                "id": "back",
              },
            ],
            "value": "React",
          },
          Object {
            "id": "angular",
            "value": "Angular",
          },
          Object {
            "id": "vuejs",
            "value": "VueJS",
          },
        ],
        "id": "front1",
        "title": "foo",
      },
      Object {
        "cells": Array [
          Object {
            "id": "python",
            "value": "Python",
          },
          Object {
            "id": "Go",
            "subItems": Array [],
            "value": "Go",
          },
          Object {
            "id": "Java",
            "value": "java",
          },
        ],
        "id": "back",
      },
    ]
  }
  selectedCells={Object {}}
/>
`;

exports[`Table component should render a Table with 3 levels 1`] = `
<ElementaryTable
  elevatedColumnIndexes={Object {}}
  elevatedRowIndexes={Object {}}
  id="table-foo"
  indexesMapping={
    Object {
      "absolute": Object {
        "0": Object {
          "index": 0,
          "parentIndex": null,
        },
        "1": Object {
          "index": 1,
          "parentIndex": null,
        },
        "2": Object {
          "index": 2,
          "parentIndex": null,
        },
      },
      "relative": Object {
        "0": Object {
          "index": 0,
        },
        "1": Object {
          "index": 1,
        },
        "2": Object {
          "index": 2,
        },
      },
    }
  }
  onRowClose={[Function]}
  onRowOpen={[Function]}
  openedTrees={Object {}}
  rows={
    Array [
      Object {
        "cells": Array [
          Object {
            "id": "wawoo",
            "value": "Wawooo!",
          },
          Object {
            "id": "wah",
            "value": "Wah!",
          },
          Object {
            "id": "cool",
            "value": "Cool!",
          },
        ],
        "id": "front",
        "isHeader": true,
        "isSelectable": false,
      },
      Object {
        "cells": Array [
          Object {
            "id": "react",
            "subItems": Array [
              Object {
                "cells": Array [
                  Object {
                    "id": "storybook",
                    "value": "Storybook",
                  },
                  Object {
                    "id": "jest",
                    "value": "Jest",
                  },
                  Object {
                    "id": "Enzym",
                    "subItems": Array [
                      Object {
                        "cells": Array [
                          Object {
                            "id": "sushi",
                            "value": "Sushi",
                          },
                          Object {
                            "id": "pizza",
                            "value": "Pizza",
                          },
                          Object {
                            "id": "mafe",
                            "value": "Mafe",
                          },
                        ],
                        "id": "miam",
                      },
                    ],
                    "value": "enzyme",
                  },
                ],
                "id": "foo",
              },
              Object {
                "cells": Array [
                  Object {
                    "id": "docker",
                    "value": "Docker",
                  },
                  Object {
                    "id": "doccomp",
                    "value": "Docker compose",
                  },
                  Object {
                    "id": "Sentry",
                    "subItems": Array [
                      Object {
                        "cells": Array [
                          Object {
                            "id": "sushi",
                            "value": "Sushi",
                          },
                          Object {
                            "id": "pizza",
                            "value": "Pizza",
                          },
                          Object {
                            "id": "mafe",
                            "value": "Mafe",
                          },
                        ],
                        "id": "miam",
                      },
                    ],
                    "value": "Sentry",
                  },
                ],
                "id": "back",
              },
            ],
            "value": "React",
          },
          Object {
            "id": "angular",
            "value": "Angular",
          },
          Object {
            "id": "vuejs",
            "value": "VueJS",
          },
        ],
        "id": "front1",
        "title": "foo",
      },
      Object {
        "cells": Array [
          Object {
            "id": "python",
            "value": "Python",
          },
          Object {
            "id": "Go",
            "subItems": Array [],
            "value": "Go",
          },
          Object {
            "id": "Java",
            "value": "java",
          },
        ],
        "id": "back",
      },
    ]
  }
  selectedCells={Object {}}
/>
`;

exports[`Table component should render a Table with spans 1`] = `
<ElementaryTable
  elevatedColumnIndexes={Object {}}
  elevatedRowIndexes={Object {}}
  id="table-foo"
  indexesMapping={
    Object {
      "absolute": Object {
        "0": Object {
          "index": 0,
          "parentIndex": null,
        },
        "1": Object {
          "index": 1,
          "parentIndex": null,
        },
        "2": Object {
          "index": 2,
          "parentIndex": null,
        },
      },
      "relative": Object {
        "0": Object {
          "index": 0,
        },
        "1": Object {
          "index": 1,
        },
        "2": Object {
          "index": 2,
        },
      },
    }
  }
  isSpan={true}
  onRowClose={[Function]}
  onRowOpen={[Function]}
  openedTrees={Object {}}
  rows={
    Array [
      Object {
        "cells": Array [
          Object {
            "id": "wawoo",
            "value": "Wawooo!",
          },
          Object {
            "id": "wah",
            "value": "Wah!",
          },
          Object {
            "id": "cool",
            "value": "Cool!",
          },
        ],
        "id": "front",
        "isHeader": true,
        "isSelectable": false,
      },
      Object {
        "cells": Array [
          Object {
            "id": "react",
            "subItems": Array [
              Object {
                "cells": Array [
                  Object {
                    "id": "storybook",
                    "value": "Storybook",
                  },
                  Object {
                    "id": "jest",
                    "value": "Jest",
                  },
                  Object {
                    "id": "Enzym",
                    "subItems": Array [],
                    "value": "enzyme",
                  },
                ],
                "id": "foo",
              },
              Object {
                "cells": Array [
                  Object {
                    "id": "docker",
                    "value": "Docker",
                  },
                  Object {
                    "id": "doccomp",
                    "value": "Docker compose",
                  },
                  Object {
                    "id": "Sentry",
                    "subItems": Array [],
                    "value": "Sentry",
                  },
                ],
                "id": "back",
              },
            ],
            "value": "React",
          },
          Object {
            "id": "angular",
            "value": "Angular",
          },
          Object {
            "id": "vuejs",
            "value": "VueJS",
          },
        ],
        "id": "front1",
        "title": "foo",
      },
      Object {
        "cells": Array [
          Object {
            "id": "python",
            "value": "Python",
          },
          Object {
            "id": "Go",
            "subItems": Array [
              Object {
                "cells": Array [
                  Object {
                    "id": "storybook",
                    "value": "Storybook",
                  },
                  Object {
                    "id": "jest",
                    "value": "Jest",
                  },
                  Object {
                    "id": "Enzym",
                    "subItems": Array [],
                    "value": "enzyme",
                  },
                ],
                "id": "foo",
              },
              Object {
                "cells": Array [
                  Object {
                    "id": "docker",
                    "value": "Docker",
                  },
                  Object {
                    "id": "doccomp",
                    "value": "Docker compose",
                  },
                  Object {
                    "id": "Sentry",
                    "subItems": Array [],
                    "value": "Sentry",
                  },
                ],
                "id": "back",
              },
            ],
            "value": "Go",
          },
          Object {
            "id": "Java",
            "value": "java",
          },
        ],
        "id": "back",
      },
    ]
  }
  selectedCells={Object {}}
/>
`;

exports[`Table component should render a responsive Table 1`] = `
<ResponsiveContainer>
  [Function]
</ResponsiveContainer>
`;

exports[`Table component should render a selectable Table 1`] = `
<SelectionHandler>
  [Function]
</SelectionHandler>
`;

exports[`Table component should render a virtualized Table 1`] = `
<Virtualizer
  columnsLength={3}
  fixedCellsHeight={
    Object {
      "count": 0,
      "sum": 0,
    }
  }
  fixedCellsWidth={
    Object {
      "count": 0,
      "sum": 0,
    }
  }
  fixedColumns={Array []}
  fixedRows={Array []}
  height={300}
  hiddenColumns={Array []}
  hiddenRows={Array []}
  horizontalPadding={0}
  initialScroll={
    Object {
      "columnIndex": 3,
      "rowIndex": 4,
    }
  }
  rowsLength={3}
  verticalPadding={0}
  width={400}
>
  [Function]
</Virtualizer>
`;

exports[`Table component should render the default table 1`] = `
<ElementaryTable
  elevatedColumnIndexes={Object {}}
  elevatedRowIndexes={Object {}}
  id="table-foo"
  indexesMapping={
    Object {
      "absolute": Object {
        "0": Object {
          "index": 0,
          "parentIndex": null,
        },
        "1": Object {
          "index": 1,
          "parentIndex": null,
        },
        "2": Object {
          "index": 2,
          "parentIndex": null,
        },
        "3": Object {
          "index": 3,
          "parentIndex": null,
        },
        "4": Object {
          "index": 4,
          "parentIndex": null,
        },
        "5": Object {
          "index": 5,
          "parentIndex": null,
        },
      },
      "relative": Object {
        "0": Object {
          "index": 0,
        },
        "1": Object {
          "index": 1,
        },
        "2": Object {
          "index": 2,
        },
        "3": Object {
          "index": 3,
        },
        "4": Object {
          "index": 4,
        },
        "5": Object {
          "index": 5,
        },
      },
    }
  }
  onRowClose={[Function]}
  onRowOpen={[Function]}
  openedTrees={Object {}}
  rows={
    Array [
      Object {
        "cells": Array [
          Object {
            "id": "wawoo",
            "value": "Wawooo!",
          },
          Object {
            "id": "wah",
            "value": "Wah!",
          },
          Object {
            "id": "cool",
            "value": "Cool!",
          },
        ],
        "id": "front",
        "isHeader": true,
      },
      Object {
        "cells": Array [
          Object {
            "cellContent": undefined,
            "cellContentProps": undefined,
            "id": "react",
            "value": "React",
          },
          Object {
            "id": "angular",
            "value": "Angular",
          },
          Object {
            "id": "vuejs",
            "value": "VueJS",
          },
        ],
        "id": "front1",
      },
      Object {
        "cells": Array [
          Object {
            "id": "python",
            "value": "Python",
          },
          Object {
            "id": "Go",
            "value": "Go",
          },
          Object {
            "id": "Java",
            "value": "java",
          },
        ],
        "id": "back",
      },
      Object {
        "cells": Array [
          Object {
            "id": "python",
            "value": "Python1",
          },
          Object {
            "id": "Go",
            "value": "Go",
          },
          Object {
            "id": "Java",
            "value": "java",
          },
        ],
        "id": "back1",
      },
      Object {
        "cells": Array [
          Object {
            "id": "python",
            "value": "Python2",
          },
          Object {
            "id": "Go",
            "value": "Go",
          },
          Object {
            "id": "Java",
            "value": "java",
          },
        ],
        "id": "back2",
      },
      Object {
        "cells": Array [
          Object {
            "id": "python",
            "value": "Python3",
          },
          Object {
            "id": "Go",
            "value": "Go",
          },
          Object {
            "id": "Java",
            "value": "java",
          },
        ],
        "id": "back3",
      },
    ]
  }
  selectedCells={Object {}}
/>
`;
